File: tests/fixtures/generic/clean/taint_xpathi_clean.php
<?php
// @audit-fixture
// @rule: taint_xpathi
// @category: security
// @expected: clean
$dom = new DOMDocument();
$dom->loadXML($xmlData);
$xpath = new DOMXPath($dom);
$name = preg_replace('/[^\w]/', '', $_GET['name']);
$results = $xpath->query("//user[@name='$name']");

File: tests/fixtures/generic/clean/taint_xpathi_clean.java
// @audit-fixture
// @rule: taint_xpathi
// @category: security
// @expected: clean
import javax.xml.xpath.*;
public class XPathServlet extends HttpServlet {
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws Exception {
        // Utiliser XPathVariableResolver pour des requêtes paramétrées
        String name = request.getParameter("name").replaceAll("[^\\w]", "");
        String result = xpath.evaluate("//user[@name='" + name + "']", doc);
    }
}
